drivers/rtc/rtc_rx8130ce.c

```diff
@@ -0,0 +1,758 @@
+		}
+	}
+	LOG_DBG("get calibration: offset = 0x%02x, freq_ppb = %d", offset, *freq_ppb);
+
+exit:
+	k_mutex_unlock(&data->lock);
+	return rc;
+}
+#endif /* CONFIG_RTC_CALIBRATION */
+
+
+static int rx8130ce_init(const struct device *dev)
+{
+	int rc;
+	const struct rx8130ce_config *cfg = dev->config;
+	struct rx8130ce_data *data = dev->data;
+
+	data->dev = dev;
+	k_mutex_init(&data->lock);
+	if (!i2c_is_ready_dt(&cfg->i2c)) {
+		LOG_ERR("I2C bus not ready");
+		return -ENODEV;
+	}
+
+	/* read all control registers */
+	rc = i2c_burst_read_dt(&cfg->i2c, EXTENSION, (uint8_t *)&data->reg, sizeof(data->reg));
+	if (rc != 0) {
+		LOG_ERR("Failed to read control registers");
+		return rc;
+	}
+	data->reg.flag = 0x00;
+	data->reg.extension &= ~EXT_TE;
+
+
+	switch (cfg->clockout_frequency) {
+	case 0: /* OFF */
+		data->reg.extension |= EXT_FSEL1 | EXT_FSEL0;
+		break;
+	case 1: /* 1 Hz */
+		data->reg.extension &= ~EXT_FSEL0;
+		data->reg.extension |= EXT_FSEL1;
+		break;
+	case 2: /* 1.024 kHz */
+		data->reg.extension |= EXT_FSEL0;
+		data->reg.extension &= ~EXT_FSEL1;
+		break;
+	case 3: /* 32.768 kHz */
+		data->reg.extension &= ~(EXT_FSEL1 | EXT_FSEL0);
+		break;
+	default:
+		LOG_ERR("Invalid clockout frequency option: %d", cfg->clockout_frequency);
+		return -EINVAL;
+	}
+
+	if (cfg->battery_switchover != 0) {
+		/* Enable initial voltage detection, following settings depend
+		 * on if the CTRL1_INIEN has been set prior (lifetime)
+		 */
+		data->reg.ctrl1 |= CTRL1_INIEN;
+		rc = i2c_burst_write_dt(&cfg->i2c, CTRL1,
+			  (uint8_t *)&data->reg.ctrl1, sizeof(data->reg.ctrl1));
+		if (rc != 0) {
+			LOG_ERR("Failed to write ctrl1 register");
+			return rc;
+		}
+	}
+
+	switch (cfg->battery_switchover) {
+	case 1: /* Power switch on, non rechargeable battery */
+		data->reg.ctrl1 |= CTRL1_INIEN;
+		break;
+	case 2: /* Power switch on, rechargeable battery */
+		data->reg.ctrl1 &= ~(CTRL1_INIEN | CTRL1_CHGEN);
+		break;
+	case 3: /* Power switch on, rechargeable battery, i2c & Fout disabled if VDD < Vdet1 */
+		data->reg.ctrl1 |= CTRL1_CHGEN | CTRL1_INIEN;
+		break;
+	case 4: /* Power switch on, rechargeable battery, i2c & Fout always enabled */
+		data->reg.ctrl1 |= CTRL1_CHGEN;
+		data->reg.ctrl1 &= ~CTRL1_INIEN;
+		break;
+	}
+
+#if defined(CONFIG_RTC_ALARM) || defined(CONFIG_RTC_UPDATE)
+	k_work_init(&data->irq_work, rx8130ce_irq_work_handler);
+	if (cfg->irq.port != NULL) {
+		gpio_pin_configure_dt(&cfg->irq, GPIO_INPUT);
+		gpio_init_callback(&data->irq_cb, rx8130ce_irq, BIT(cfg->irq.pin));
+		rc = gpio_add_callback_dt(&cfg->irq, &data->irq_cb);
+		if (rc != 0) {
+			LOG_ERR("Failed to add callback");
+			return rc;
+		}
+	}
+#endif /* CONFIG_RTC_ALARM || CONFIG_RTC_UPDATE */
+	rc = i2c_burst_write_dt(&cfg->i2c, EXTENSION, (uint8_t *)&data->reg, sizeof(data->reg));
+	if (rc != 0) {
+		LOG_ERR("Failed to write control registers");
+		return rc;
+	}
+	return 0;
+}
+
+static const struct rtc_driver_api rx8130ce_driver_api = {
```

Comment:
Wrap using the API macros

```suggestion
static DEVICE_API(rtc, rx8130ce_driver_api) = {
```